### PR TITLE
Initial test framework for emitter

### PR DESCRIPTION
**Pull request by dolauli**
Following are steps to run tests:

1. Go to `tests-upgrade\tests-emitter`
2. Run `npm install` to install dependencies
3. Run `.\EmitterTest.ps1 -AllowList` to run tests

### CODE REVIEW

**Review by VeryEarly — COMMENTED** [review]
how about download tsp from spec repo instead of copy over?

File: tests-upgrade/tests-emitter/Sphere.Management/target/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+<!-- region Generated -->
```
Comment (VeryEarly) [inline tests-upgrade/tests-emitter/Sphere.Management/target/README.md:1]: does target folder need to be checked-in?

Comment (dolauli) [reply tests-upgrade/tests-emitter/Sphere.Management/target/README.md:1]: Yes, it is manually maintained.

Comment (VeryEarly) [reply tests-upgrade/tests-emitter/Sphere.Management/target/README.md:1]: > Yes, it is manually maintained.

isn't it output of tsp compile --emit ? why the generated content need to be checked-in?

Comment (dolauli) [reply tests-upgrade/tests-emitter/Sphere.Management/target/README.md:1]: The output of tsp compile will be put in **generated** folder. **Target** folder contains the expected code.

**Pull request merged**